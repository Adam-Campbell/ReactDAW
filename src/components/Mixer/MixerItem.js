import React from 'react';
import PropTypes from 'prop-types';
import Meter from './Meter';

const MixerItem = props => (
    <div className="mixer__item">
        <div className="mixer__slider-container">
            <div className="mixer__volume-marker-container">
                <p className="mixer__volume-marker">10</p>
                <p className="mixer__volume-marker">0</p>
                <p className="mixer__volume-marker">-10</p>
                <p className="mixer__volume-marker">-20</p>
                <p className="mixer__volume-marker">-30</p>
                <p className="mixer__volume-marker">-40</p>
            </div>
            <input
                className="range-input range-input__mixer-volume"
                type="range"
                id="volume-slider"
                min={-40}
                max={10}
                step={0.25}
                value={props.volume}
                onChange={props.handleVolumeChange}
            ></input>
            <Meter trackId={props.trackId} />
        </div>
        <div className="mixer__button-container">
            <button 
                className={`mixer__button ${props.isMuted ? 'mixer__button--active' : ''}`}
                onClick={props.handleMuteButtonClick}
            >M</button>
            <button 
                className={`mixer__button ${props.isSolo ? 'mixer__button--active' : ''}`}
                onClick={props.handleSoloButtonClick}
            >S</button>
        </div>
        <input
            className="range-input range-input__mixer-pan"
            type="range"
            id="panner-slider"
            min={-1}
            max={1}
            step={0.005}
            value={props.pan}
            onChange={props.handlePanChange}
        ></input>
        <div className="mixer__pan-label-container">
            <p className="mixer__pan-label">L</p>
            <p className="mixer__pan-label">R</p>
        </div>
        <p className="mixer__track-name">{props.name}</p>
    </div>
);

MixerItem.propTypes = {
    volume: PropTypes.number.isRequired,
    handleVolumeChange: PropTypes.func.isRequired,
    trackId: PropTypes.string.isRequired,
    handleMuteButtonClick: PropTypes.func.isRequired,
    handleSoloButtonClick: PropTypes.func.isRequired,
    pan: PropTypes.number.isRequired,
    handlePanChange: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    isMuted: PropTypes.bool.isRequired,
    isSolo: PropTypes.bool.isRequired
};

export default MixerItem;