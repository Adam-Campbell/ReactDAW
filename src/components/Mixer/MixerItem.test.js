import React from 'react';
import { shallow } from 'enzyme';
import MixerItem from './MixerItem';

test('renders correctly', () => {
    const component = shallow(
        <MixerItem 
            volume={0}
            pan={0}
            isMuted={false}
            isSolo={false}
            trackId="2812513466104156"
            name="Channel 1"
            handleVolumeChange={jest.fn()}
            handleMuteButtonClick={jest.fn()}
            handleSoloButtonClick={jest.fn()}
            handlePanChange={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
});