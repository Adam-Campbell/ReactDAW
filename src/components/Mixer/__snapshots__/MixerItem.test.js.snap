// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mixer__item"
>
  <div
    className="mixer__slider-container"
  >
    <div
      className="mixer__volume-marker-container"
    >
      <p
        className="mixer__volume-marker"
      >
        10
      </p>
      <p
        className="mixer__volume-marker"
      >
        0
      </p>
      <p
        className="mixer__volume-marker"
      >
        -10
      </p>
      <p
        className="mixer__volume-marker"
      >
        -20
      </p>
      <p
        className="mixer__volume-marker"
      >
        -30
      </p>
      <p
        className="mixer__volume-marker"
      >
        -40
      </p>
    </div>
    <input
      className="range-input range-input__mixer-volume"
      id="volume-slider"
      max={10}
      min={-40}
      onChange={[MockFunction]}
      step={0.25}
      type="range"
      value={0}
    />
    <Meter
      trackId="2812513466104156"
    />
  </div>
  <div
    className="mixer__button-container"
  >
    <button
      className="mixer__button "
      onClick={[MockFunction]}
    >
      M
    </button>
    <button
      className="mixer__button "
      onClick={[MockFunction]}
    >
      S
    </button>
  </div>
  <input
    className="range-input range-input__mixer-pan"
    id="panner-slider"
    max={1}
    min={-1}
    onChange={[MockFunction]}
    step={0.005}
    type="range"
    value={0}
  />
  <div
    className="pan__label-container"
  >
    <p
      className="pan__label"
    >
      L
    </p>
    <p
      className="pan__label"
    >
      R
    </p>
  </div>
  <p
    className="mixer__track-name"
  >
    Channel 1
  </p>
</div>
`;
